src/main/java/org/logstash/beats/Server.java

```diff
@@ -51,14 +54,15 @@ public Server listen() throws InterruptedException {
                 logger.error("Could not shut down worker group before starting", e);
             }
         }
+        bossGroup = new NioEventLoopGroup(eventLoopThreadCount); // TODO: add a config to make it adjustable, no need many threads
         workGroup = new NioEventLoopGroup(eventLoopThreadCount);
         try {
             logger.info("Starting server on port: {}", this.port);
 
             beatsInitializer = new BeatsInitializer(messageListener, clientInactivityTimeoutSeconds, executorThreadCount);
 
             ServerBootstrap server = new ServerBootstrap();
-            server.group(workGroup)
+            server.group(bossGroup, workGroup)
```

Comment:
Till now, we were using workgroup for handling incoming socket connections and there was no guarantee socket handlers got terminated first. With this separation we will have two major improvements:
- make sure we terminate handlers, which don't push to netty internal tasks queue
- make thread count configurable: no need many threads as it accepts and sends to work group

References:
- https://dzone.com/articles/building-a-performant-application-using-netty
- https://netty.io/wiki/user-guide-for-4.x.html#wiki-h3-16